### DISCUSSION TITLE
Installation procedure get stuck with google-guest-agent service on Ubuntu 18.04🐞

### DISCUSSION THREAD

**Discussion by gsilano** [ctu-mrs/mrs_uav_system#76] [General]
Running the automatic installation procedure on Ubuntu 18.04 over an existing release of the MRS system (for the case of interest that was that in October 2020), the system gets stuck with the `google-gust-agent` service and the installation fails. Whether trying to remove the service by using `sudo dpkg` command or the `sudo apt remove` command there is no way to fix it. The installation procedure still fails. This appears to be due to a service restarting each time is interrupted. Then, the installation stops by raising locks on any other command (e.g., `sudo apt update` or `sudo apt upgrade`). @bednarhonza can tell more about this.

Eventually, @ViktorWalter solved the issue by running `sudo htop` and killing the service and its related dependencies when it got stuck. Of course, this can be considered only as a temporary fix. Students or even me could have trouble in case a system update is required. 

Below are listed the services we stopped to make it work.

```
sudo systemctl stop google-guest-agent.service google-shutdown-scripts.service google-startup-scripts.service 
```

**Comment by spurnvoj** [comment]
This service should no be installed anymore, so new students will not have
this problem.

**Reply by ViktorWalter** [reply]
To my knowledge, this service IS installed (as a dependency of something) but it is then subsequently uninstalled. Each of these operations is stopped on @gsilano 's PC by the previously running service. I'm not sure if the service is allowed to start between the two actions on a fresh installation.

**Reply by gsilano** [reply]
This command helps fix the issue:

```
sudo apt remove --purge google-guest-agent google-compute-engine google-compute-engine-oslogin
```

If the uninstall process should get stuck, run `htop`, search for the `google-guest` service, and kill it. After that, it should move on with any problems.

**Reply by matemat13** [reply]
Hmm the strange thing is that none of our stuff seems to be installing anything google-related. It's a mystery how it always gets installed in the first place. The only thing google related I could find in our code is in `uav_modules`: The realsense driver installs `libgoogle-glog-dev` (doesn't depend on `gce` or anything related AFAIK) and `uvdar_core` and `uvdar_gazebo_plugin` both install the `gce-compute-image-packages`, but only in CI.
